src/main/java/uk/nhs/hee/tis/trainee/forms/service/FormRPartAService.java

```diff
@@ -73,15 +76,27 @@ public class FormRPartAService {
    * @param repository            spring data repository
    * @param cloudObjectRepository repository to storage form in the cloud
    * @param mapper                maps between the form entity and dto
+   * @param objectMapper          The object mapper.
+   * @param traineeIdentity       The trainee identity.
    */
   public FormRPartAService(FormRPartARepository repository,
       S3FormRPartARepositoryImpl cloudObjectRepository,
       FormRPartAMapper mapper,
-      ObjectMapper objectMapper) {
+      ObjectMapper objectMapper, TraineeIdentity traineeIdentity) {
     this.repository = repository;
     this.cloudObjectRepository = cloudObjectRepository;
     this.mapper = mapper;
     this.objectMapper = objectMapper;
+    this.traineeIdentity = traineeIdentity;
+  }
+
+  /**
+   * Get the logged-in trainee's ID.
+   *
+   * @return The trainee TIS ID.
+   */
+  public String getLoggedInTraineeId() {
+    return traineeIdentity.getTraineeId();
```

Comment:
There are a couple of better options for this
1. Inject the TraineeIdentity in to the resource as well the as the service, so we don't need to retrieve it separately.
2. Mimic trainee detail's [NotRecordOwnerException ](https://github.com/Health-Education-England/tis-trainee-details/blob/c139b01f5f3aa653df7d59ab5cec325cb1c6993f/src/main/java/uk/nhs/hee/trainee/details/exception/NotRecordOwnerException.java)and move the ID verification to the service class.